src/main/java/edu/ucdavis/dss/ipa/api/components/assignment/AssignmentViewTeachingAssignmentController.java

```diff
@@ -25,19 +25,21 @@
     @Inject DataWarehouseRepository dwRepository;
     @Inject Authorizer authorizer;
     @Inject BudgetScenarioService budgetScenarioService;
+    @Inject InstructorTypeService instructorTypeService;
+    @Inject UserRoleService userRoleService;
 
     @RequestMapping(value = "/api/assignmentView/schedules/{scheduleId}/teachingAssignments", method = RequestMethod.POST, produces="application/json")
     @ResponseBody
     public TeachingAssignment addTeachingAssignment(@PathVariable long scheduleId, @RequestBody TeachingAssignment teachingAssignment, HttpServletResponse httpResponse) {
         Instructor instructor = null;
+        InstructorType instructorType = null;
 
         if (teachingAssignment != null && teachingAssignment.getInstructor() != null) {
             instructor = instructorService.getOneById(teachingAssignment.getInstructor().getId());
         }
 
-        if (instructor == null) {
-            httpResponse.setStatus(HttpStatus.BAD_REQUEST.value());
-            return null;
+        if (teachingAssignment != null && teachingAssignment.getInstructorType() != null) {
```

Comment:
It looks like this method requires teachingAssignment to be non-null. Check for null at the top of the method and return immediately with an error if it is null, then remove all the remaining null checks.